src/abstractdataframe/iteration.jl

```diff
@@ -29,16 +38,42 @@ Base.getindex(itr::DFRowIterator, i) = DataFrameRow(itr.df, i)
 Base.map(f::Function, dfri::DFRowIterator) = [f(row) for row in dfri]
 
 # Iteration by columns
+"""
+    DFColumnIterator{<:AbstractDataFrame}
+
+Iterator over columns of an `AbstractDataFrame`.
+Each returned value is a tuple consisting of column name and column data.
+
+A value of this type is returned by `eachcol` function.
+"""
 struct DFColumnIterator{T <: AbstractDataFrame}
     df::T
 end
+
+"""
+    eachcol(::AbstractDataFrame)
+
+Return `DFColumnIterator` that iterates an `AbstractDataFrame` column by column.
+Iteration returns a tuple consisting of column name and column data.
+
+`DFColumnIterator` has a custom implementation of the `map` function which
+returns a `DataFrame` and assumes that a function argument passed do
+the `map` function accepts only column data.
+
+**Examples**
+
+```julia
+df = DataFrame(x=1:4, y=11:14)
+map(sum, eachcol(df)) == DataFrame(x=10, y=50)
```

Comment:
Better use a `jldoctest` and copy the REPL output.

Reply:
fixed